runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1175,3 +1176,163 @@ J9::ValuePropagation::getParmValues()
 
    TR_ASSERT(parmIterator->atEnd() && parmIndex == numParms, "Bad signature for owning method");
    }
+
+static bool isTakenSideOfAVirtualGuard(TR::Block* block)
```

Comment:
This seems misleading - what about an OSR guard or a virtual guard which is implemented using OSR... Why not just check if the block's predecessors to see if the end in a branch that is a virtual guard?

Reply:
Changed to check the last tree of the block's predecessor.